jpype/_core.py

```diff
@@ -231,24 +241,51 @@ def startJVM(
         jvmpath = os.fspath(jvmpath)
 
     # Classpath handling
-    if _hasClassPath(jvmargs):
+    old_classpath = _getOldClassPath(jvmargs)
+    if old_classpath:
         # Old style, specified in the arguments
         if classpath is not None:
             # Cannot apply both styles, conflict
             raise TypeError('classpath specified twice')
+        classpath = old_classpath
     elif classpath is None:
         # Not specified at all, use the default classpath.
         classpath = _classpath.getClassPath()
 
     # Handle strings and list of strings.
-    extra_jvm_args = []
-    if classpath:
-        cp = _classpath._SEP.join(_handleClassPath(classpath))
-        extra_jvm_args += ['-Djava.class.path=%s'%cp ]
+    extra_jvm_args: list[str] = []
 
     supportLib = os.path.join(os.path.dirname(os.path.dirname(__file__)), "org.jpype.jar")
     if not os.path.exists(supportLib):
         raise RuntimeError("Unable to find org.jpype.jar support library at " + supportLib)
+
+    late_load = not has_classloader
+    if classpath:
+        cp = _classpath._SEP.join(_handleClassPath(classpath))
+        if cp.isascii():
+            # no problems
+            extra_jvm_args += ['-Djava.class.path=%s'%cp ]
+            jvmargs = _removeClassPath(jvmargs)
+            late_load = False
+        elif has_classloader:
+            # https://bugs.openjdk.org/browse/JDK-8079633?jql=text%20~%20%22ParseUtil%22
+            raise ValueError("system classloader cannot be specified with non ascii characters in the classpath")
+        elif supportLib.isascii():
+            # ok, setup the jpype system classloader and add to the path after startup
+            # this guarentees all classes have the same permissions as they did in the past
+            extra_jvm_args += [
+                '-Djava.system.class.loader=org.jpype.classloader.JpypeSystemClassLoader',
+                '-Djava.class.path=%s'%supportLib
+            ]
+            jvmargs = _removeClassPath(jvmargs)
```

Comment:
Should we pass the class path as a different variable and then use it.   The issue I am concerned about is that of those jars that specify a service.

https://medium.com/@yureshcs/services-and-service-provider-with-java-9-modules-ddd0170749b0

In that past when we did testing is was clear that services were only recognized if they were loaded on the initial class_path.   I believe it is the case for certain things like databases.   Unfortunately, it was a while since I last looked at the issue so I don't have a working test case.